src/test/java/com/teragrep/lsh_01/RegexPayloadTest.java

```diff
@@ -0,0 +1,87 @@
+/*
+  logstash-http-input to syslog bridge
+  Copyright 2024 Suomen Kanuuna Oy
+
+  Derivative Work of Elasticsearch
+  Copyright 2012-2015 Elasticsearch
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+package com.teragrep.lsh_01;
+
+import com.teragrep.lsh_01.conversion.DefaultPayload;
+import com.teragrep.lsh_01.conversion.RegexPayload;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+public class RegexPayloadTest {
+
+    @Test
+    public void testDefaultSplitRegex() {
+        String body = "foo\nbar\nfoobar";
+        Pattern splitPattern = Pattern.compile("\\n");
+        RegexPayload payload = new RegexPayload(new DefaultPayload(body), splitPattern);
+        List<String> messages = payload.messages();
+
+        Assertions.assertEquals(3, messages.size());
+        Assertions.assertEquals("foo", messages.get(0));
+        Assertions.assertEquals("bar", messages.get(1));
+        Assertions.assertEquals("foobar", messages.get(2));
+    }
+
+    @Test
+    public void testCustomSplitRegex() {
+        String body = "foo,bar,foobar";
+        Pattern splitPattern = Pattern.compile(",");
+        RegexPayload payload = new RegexPayload(new DefaultPayload(body), splitPattern);
+        List<String> messages = payload.messages();
+
+        Assertions.assertEquals(3, messages.size());
+        Assertions.assertEquals("foo", messages.get(0));
+        Assertions.assertEquals("bar", messages.get(1));
+        Assertions.assertEquals("foobar", messages.get(2));
+    }
+
+    @Test
+    public void testNoSplittingRequired() {
+        String body = "foobar";
+        Pattern splitPattern = Pattern.compile("\\n");
+        RegexPayload payload = new RegexPayload(new DefaultPayload(body), splitPattern);
+        List<String> messages = payload.messages();
+
+        Assertions.assertEquals(1, messages.size());
+        Assertions.assertEquals("foobar", messages.get(0));
+    }
+
+    @Test
+    public void testObjectEquality() {
```

Comment:
i would separate this into two tests, equals and not equals

Reply:
Split into two tests.